### PR TITLE
feat(gaussdb/nosql): add new resource influx and mongo

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new resource influx and mongo

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add new resource influx and mongo
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussInfluxInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussInfluxInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussInfluxInstance_basic
=== PAUSE TestAccGaussInfluxInstance_basic
=== CONT  TestAccGaussInfluxInstance_basic
--- PASS: TestAccGaussInfluxInstance_basic (2290.99s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb      2291.310s

make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run TestAccGaussMongoInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run TestAccGaussMongoInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussMongoInstance_basic
=== PAUSE TestAccGaussMongoInstance_basic
=== CONT  TestAccGaussMongoInstance_basic
--- PASS: TestAccGaussMongoInstance_basic (1406.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1406.397s
```

### CODE REVIEW

File: docs/resources/gaussdb_influx_instance.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+* `keep_days` - (Optional, Int) Specifies the number of days to retain the generated backup files. The value ranges from
+  **0** to **35**. If this parameter is set to **0**, the automated backup policy is not set. If this parameter is not
+  transferred, the automated backup policy is enabled by default. Backup files are stored for seven days by default.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Specifies a resource ID in UUID format.
```
Comment (hmbig2) [inline docs/resources/gaussdb_influx_instance.md]: s/Specifies/Indicates

File: docs/resources/gaussdb_mongo_instance.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+* `keep_days` - (Optional, Int) Specifies the number of days to retain the generated backup files. The value ranges from
+  **0** to **35**. If this parameter is set to **0**, the automated backup policy is not set. If this parameter is not
+  transferred, the automated backup policy is enabled by default. Backup files are stored for seven days by default.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Specifies a resource ID in UUID format.
```
Comment (hmbig2) [inline docs/resources/gaussdb_mongo_instance.md]: ditto

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go [hunk 3]
```diff
@@ -258,7 +265,17 @@ func ResourceGeminiDBInstanceV3() *schema.Resource {
 	}
 }
 
-func resourceGeminiDBDataStore(d *schema.ResourceData) instances.DataStore {
+func resourceGaussDBCassandraInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	defaults := defaultValues{
+		Mode:      "Cluster",
+		dbType:    "cassandra",
+		dbVersion: "3.11",
```
Comment (hmbig2) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go:275]: seem all params are required, the default values are unnecessary

File: docs/resources/gaussdb_influx_instance.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/gaussdb_influx_instance.md]: fixed

File: docs/resources/gaussdb_mongo_instance.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/gaussdb_mongo_instance.md]: fixed

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go:275]: this is correct under current logic
will refactor the gaussdb resources and use the true default value way latter

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go [hunk 4]
```diff
@@ -122,7 +129,7 @@ func ResourceGeminiDBInstanceV3() *schema.Resource {
 							Required: true,
 							ForceNew: true,
 							ValidateFunc: validation.StringInSlice([]string{
-								"GeminiDB-Cassandra",
```
Comment (ShiChangkuo) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go:125]: it's better to add the two values in the slice.

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go [hunk 5]
```diff
@@ -607,7 +624,16 @@ func resourceGeminiDBInstanceV3Delete(d *schema.ResourceData, meta interface{})
 	return nil
 }
 
-func resourceGeminiDBInstanceV3Update(d *schema.ResourceData, meta interface{}) error {
+func resourceGaussDBCassandraInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	defaults := defaultValues{
+		Mode:      "Cluster",
+		dbType:    "cassandra",
+		dbVersion: "3.11",
```
Comment (ShiChangkuo) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go:634]: `defaults.logName` is empty in Update function

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go [hunk 6]
```diff
@@ -122,7 +129,7 @@ func ResourceGeminiDBInstanceV3() *schema.Resource {
 							Required: true,
 							ForceNew: true,
 							ValidateFunc: validation.StringInSlice([]string{
-								"GeminiDB-Cassandra",
+								"cassandra",
```
Comment (ShiChangkuo) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go]: docs also should be updated

Comment (Jason-Zhang9309) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go]: use "GeminiDB-Cassandra" will cause ForceNew, so we should remove it from the doc

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go:634]: fixed

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_cassandra_instance.go:125]: fixed

**Pull request merged**